File: crates/filter-list-manager/src/manager/models/active_rules_info.rs
//! Represents list of active (not disabled) rules with extra filter data
use crate::FilterId;

/// Represents list of active (not disabled) rules with extra filter data
pub struct ActiveRulesInfo {
    /// Filter id for these rules
    pub filter_id: FilterId,
    /// Group id of the filter
    pub group_id: i32,
    /// Is this filter trusted?
    pub is_trusted: bool,
    /// List of active rules.
    /// There rules is the difference between all rules of filter and disabled rules of filter
    pub rules: Vec<String>,
}

File: platform/windows/AdGuard.FilterListManager/MarshalLogic/ActiveRulesInfo.cs
using System.Collections.Generic;

namespace AdGuard.FilterListManager.MarshalLogic
{
    public class ActiveRulesInfo
    {
        public long FilterId { get; }
        public int GroupId { get; }
        public bool IsTrusted { get; }
        public List<string> Rules { get; }

        public ActiveRulesInfo(long filterId, int groupId, bool isTrusted, List<string> rules)
        {
            FilterId = filterId;
            GroupId = groupId;
            IsTrusted = isTrusted;
            Rules = rules;
        }
    }
}
